content/hi/docs/concepts/architecture/controller.md

```diff
@@ -0,0 +1,167 @@
+
+जॉब के विपरीत, कुछ नियंत्रकों को परिवर्तन करने की आवश्यकता होती है
+आपके क्लस्टर के बाहर की चीजें।
+
+उदाहरण के लिए, यदि आप यह सुनिश्चित करने के लिए नियंत्रण पाश का उपयोग करते हैं
+पर्याप्त हैं
+आपके क्लस्टर में, तो उस नियंत्रक को बाहर कुछ चाहिए
+जरूरत पड़ने पर नए नोड्स स्थापित करने के लिए वर्तमान क्लस्टर।
+
+नियंत्रक जो बाहरी राज्य के साथ बातचीत करते हैं, वे अपनी वांछित स्थिति पाते हैं
+एपीआई सर्वर, तो लाने के लिए एक बाहरी प्रणाली के साथ सीधे संवाद
+वर्तमान स्थिति लाइन के करीब।
+
+(वास्तव में एक [नियंत्रक] है (https://github.com/kubernetes/autoscaler/)
+जो क्षैतिज रूप से आपके क्लस्टर में नोड्स को मापता है।)
+
+यहां महत्वपूर्ण बिंदु यह है कि नियंत्रक लाने के लिए कुछ बदलाव करता है
+आपकी वांछित स्थिति, और फिर वर्तमान स्थिति को आपके क्लस्टर के एपीआई सर्वर पर रिपोर्ट करता है।
+अन्य नियंत्रण लूप उस रिपोर्ट किए गए डेटा को देख सकते हैं और अपनी कार्रवाई कर सकते हैं।
+
+थर्मोस्टेट उदाहरण में, यदि कमरा बहुत ठंडा है तो एक अलग नियंत्रक
+फ्रॉस्ट प्रोटेक्शन हीटर भी चालू कर सकता है। कुबेरनेट्स क्लस्टर्स के साथ, control
+विमान परोक्ष रूप से आईपी पता प्रबंधन उपकरण, भंडारण सेवाओं के साथ काम करता है,
+क्लाउड प्रदाता एपीआई, और अन्य सेवाएं
+[Kubernetes का विस्तार](/docs/concepts/extend-kubernetes/) इसे लागू करने के लिए।
+
+## वांछित बनाम वर्तमान स्थिति {#वांछित-बनाम-वर्तमान}
+
+कुबेरनेट्स सिस्टम का क्लाउड-नेटिव व्यू लेता है, और इसे संभालने में सक्षम है
+निरंतर बदलाव।
+
+जैसे-जैसे काम होता है आपका क्लस्टर किसी भी समय बदल सकता है
+नियंत्रण लूप स्वचालित रूप से विफलताओं को ठीक करते हैं। इस का मतलब है कि,
+संभावित रूप से, आपका क्लस्टर कभी भी स्थिर स्थिति में नहीं पहुंचता है।
+
+जब तक आपके क्लस्टर के नियंत्रक चल रहे हैं और बनाने में सक्षम हैं
+उपयोगी परिवर्तन, इससे कोई फर्क नहीं पड़ता कि समग्र स्थिति स्थिर है या नहीं।
+
+## डिज़ाइन
+
+अपने डिजाइन के सिद्धांत के रूप में, कुबेरनेट्स बहुत सारे नियंत्रकों का उपयोग करता है जिन्हें प्रत्येक प्रबंधित करता है
+क्लस्टर राज्य का एक विशेष पहलू। आमतौर पर, एक विशेष नियंत्रण पाश
+(नियंत्रक) एक प्रकार के संसाधन को अपनी वांछित स्थिति के रूप में उपयोग करता है, और एक अलग होता है
+उस प्रकार का संसाधन जो वांछित अवस्था को घटित करने के लिए प्रबंधन करता है। उदाहरण के लिए,
+जॉब्स के लिए एक नियंत्रक जॉब ऑब्जेक्ट्स (नए काम की खोज करने के लिए) और पॉड ऑब्जेक्ट्स को ट्रैक करता है
+(नौकरियां चलाने के लिए, और फिर यह देखने के लिए कि काम कब पूरा हुआ)। इस मामले में
+कुछ और जॉब्स बनाता है, जबकि जॉब कंट्रोलर पॉड्स बनाता है।
+
+नियंत्रण के एक, अखंड सेट के बजाय सरल नियंत्रक होना उपयोगी है
+लूप जो आपस में जुड़े हुए हैं। नियंत्रक विफल हो सकते हैं, इसलिए कुबेरनेट्स को इसके लिए डिज़ाइन किया गया है
+उसके लिए अनुमति दें।
+
+{{< नोट >}}
+कई नियंत्रक हो सकते हैं जो एक ही प्रकार की वस्तु बनाते या अद्यतन करते हैं।
+पर्दे के पीछे, कुबेरनेट्स नियंत्रक सुनिश्चित करते हैं कि वे केवल ध्यान दें
+उनके नियंत्रण संसाधन से जुड़े संसाधनों के लिए।
+
+उदाहरण के लिए, आपके पास तैनाती और नौकरियां हो सकती हैं; ये दोनों पॉड्स बनाते हैं।
+जॉब कंट्रोलर आपके परिनियोजन द्वारा बनाए गए पॉड्स को नहीं हटाता है,
+क्योंकि जानकारी है ({{< glosary_tooltip term_id="label" text="labels" >}})
```

Comment:
Typo in the `glossary_tooltip` shortcode name. Also, localize the "text" parameter passed to the shortcode to Hindi.